src/Transport/Receiving/MessagePump.cs

```diff
@@ -169,7 +173,7 @@ async Task InnerProcessMessage(Task<ServiceBusReceivedMessage> receiveTask)
             {
                 // Can happen during endpoint shutdown
             }
-            catch (Exception exception)
+            catch (Exception exception) when (!exception.IsCausedBy(messageReceivingCancellationToken))
```

Comment:
In other message pumps we only use this "when" when catching `OperationCancelled` exception?

Reply:
On master: it is on Exception: https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/blob/master/src/Transport/Receiving/MessagePump.cs#L211 

As the IsCausedBy is an extension method on Exception that checks if it is OperationCancelled: https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/blob/master/src/Transport/ExceptionExtensions.cs#L9